config/src/config/api_config.rs

```diff
@@ -148,11 +148,13 @@ impl ConfigSanitizer for ApiConfig {
         }
 
         // Verify that failpoints are not enabled in mainnet
-        if chain_id.is_mainnet() && api_config.failpoints_enabled {
-            return Err(Error::ConfigSanitizerFailed(
-                sanitizer_name,
-                "Failpoints are not supported on mainnet nodes!".into(),
-            ));
+        if let Some(chain_id) = chain_id {
```

Comment:
there's a bunch of multi-line statements and I wonder if they'd be nicer with some kind of utility like
```Rust
fn option_chain_eq(a: Option<ChainId>, b: ChainId, none_val: bool) -> bool {
  if let Some(a_some) = a {
    a_some == b
  } else {
    none_val
  }
}
```